pkg/controller/history/controller_history.go

```diff
@@ -66,10 +67,7 @@ func NewControllerRevision(parent metav1.Object,
 	data runtime.RawExtension,
 	revision int64,
 	collisionCount *int32) (*apps.ControllerRevision, error) {
-	labelMap, err := labels.ConvertSelectorToLabelsMap(selector.String())
-	if err != nil {
-		return nil, err
-	}
+	labelMap := getPodLabelsFromData(data)
```

Comment:
I think we should require the caller of `NewControllerRevision` to provide the labels, rather than making assumptions about the contents of `data`.

cc @kow3ns @janetkuo

Reply:
Please check :)